core/src/test/scala/unit/kafka/controller/ControllerRequestMergerTest.scala

```diff
@@ -417,3 +444,12 @@ class ControllerRequestMergerTest {
     partitions
   }
 }
+
+object ControllerRequestMergerTest {
+  def testMergingDifferentStopReplicaPartitionStatesParams: java.util.stream.Stream[Arguments] = {
+    Seq(
+      Arguments.of("2.4".asInstanceOf[AnyRef], false.asInstanceOf[AnyRef]),
+      Arguments.of("3.0".asInstanceOf[AnyRef], true.asInstanceOf[AnyRef]),
+    ).asJava.stream()
+  }
+}
```

Comment:
nit: should have a new-line character at EOF (I'd recommend setting this up in IDE).